theme/layouts/_navigation.ejs

```diff
@@ -1,9 +1,25 @@
-<nav>
-  <ul>
-    <% for (const {title, link} of site.navigation) {%>
-      <li>
-        <a href="<%- link %>" target="_blank"><%- title %></a>
-      </li>
-    <% } %>
-  </ul>
+<select id="mobile-nav">
+      <% for ({title, link, sub_items} of site.navigation) { %>
+        <option value="<%- title %>"><%- title %></option>
+      <% } %>
+</select>
+
+<nav id="main-navigation">
+  <% function recNav(navig) { %>
+    <ul> 
+      <% for ({title, link, sub_items} of navig) { %>
+
+        <li>
+            <a href="<%- link %>" target="_blank"><%- title %> </a>
+              <% if (sub_items == null) {
```

Comment:
> 
> 
> This probably can be simplified, as you don't have to check here whether the item doesn't exist, also note that you need to close the tag before continuing to the next action, ex.:
> 
> ```js
> <li>
>   <a href="<%= link %>" target="_blank"><%= title %></a>
>     <% if (sub_items) { %>
>       <img src = '../images/arrow_bullet.png'/>
>       <% recNav(sub_items);
>     } %>
> </li>
> ```

this will throw error 
`navig is not iterable`
dont really know why.
Besides that we should use here `continue` to make loop jump to the next iteration if there is no sub_items.

Reply:
> we should use here continue to make loop jump to the next iteration
Having loop here, will prevent invocation of rest of the code and result in unclosed `<li>` tag, the error went unnoticed, because modern browsers are smart enough to prevent user from seeing markup error. You can observe that by clicking `view page source` in Chrome. You might see :
```
--
  <nav id="main-navigation">
  
    <ul> 
      

        <li>
            <a href="#" target="_blank">Home </a>
              

        <li>
            <a href="#" target="_blank">Documentation </a>
              
                  <img src = '../images/arrow_bullet.png'/>
                  
    <ul> 
      

        <li>
            <a href="" target="_blank">Interface </a>
              
                  <img src = '../images/arrow_bullet.png'/>
                  
    <ul> 
```
That's because of `continue` here, as it prevent the invocation of the rest of the code inside for loop. for loop will continue innovation for each item in the array irregardless of the `continue` unless there is a `return` or blocked code invocation.

Also another interesting note, you are checking `sub_items ` against, `null` and this check only passes because JavaScript is funny and `undefined == null` result in true, but `undefined === null`(check against the type) result in false. So in this case checking `if (sub_items)` may be  a better choice?

I can see some more errors in current implementation - `</ul>` being closed inside of the for loop, while being open outside.

`navig is not iterable ` more sounds like corrupted input, when `sub_items` passed to the recursion is not an array, that's an assumption, but seems like the code needs to be revisited in general considering the latest observation.

Suggestion: Ensure that the input in `config.js` is not corrupted, check small against the output and see if you get expected output or not, by observing generated HTML.